GeoLib/AnalyticalGeometry.cpp

```diff
@@ -133,11 +133,16 @@ bool lineSegmentIntersect(
 	if (parallel(v,w)) {
 		if (parallel(pq,v)) {
 			// check if c is located at v (c-a = t (b-a), t in [0,1])
-			if (qp[0] / v[0] <= 1.0)
+			if (0 <= qp[0] / v[0] && qp[0] / v[0] <= 1.0) {
```

Comment:
Can `v[0]` be 0.0?
Btw: `0 <= qp[0] / v[0]` essentially checks if ` qp[0] ` and `v[0]` have the same sign. Is that intended?